test/server/graphql/v1/collective.test.js

```diff
@@ -735,6 +768,45 @@ describe('graphql.collective.test.js', () => {
       expect(members[0].member.slug).to.equal('piamancini');
       expect(members).to.have.length(1);
     });
+
+    it('gets totalAmountSpent by collective', async () => {
+      const query = `query TotalCollectiveContributions($slug: String, $type: String) {
+        Collective(slug: $slug) {
+          id
+          currency
+          stats {
+            id
+            totalAmountSpent
+          }
+          transactions(type: $type) {
+            currency
+            netAmountInCollectiveCurrency
+          }
+        }
+      }`;
+
+      const result = await utils.graphqlQuery(query, {
+        slug: 'olu',
+        type: 'DEBIT',
+      });
+
+      result.errors && console.error(result.errors);
+      expect(result.errors).to.not.exist;
+      const collective = result.data.Collective;
+      const transactions = collective.transactions;
+      let totalAmountSpent = 0;
+
+      for (const transaction of transactions) {
+        let amount = -parseInt(transaction.netAmountInCollectiveCurrency, 10);
+        if (transaction.currency !== collective.currency) {
+          const fxRate = await getFxRate(transaction.currency, collective.currency);
+          amount = fxRate * amount;
+        }
+        totalAmountSpent = amount + totalAmountSpent;
+      }
+
+      expect(collective.stats.totalAmountSpent).to.equal(totalAmountSpent);
```

Comment:
Why not hardcode the totalAmountSpent?

Reply:
Because a slight change in the value of exchange rate will cause the test to break.

Reply:
Then I would say you need to mock the getFxRate.

In the current version, you're copying the code in the test, that's not the right approach,